presto-docs/src/main/sphinx/security/internal-communication.rst

```diff
@@ -112,3 +112,46 @@ To enable SSL/TLS for Presto internal communication, do the following:
 
         internal-communication.https.keystore.path=<keystore path>
         internal-communication.https.keystore.key=<keystore password>
+
+
+Performance with SSL/TLS enabled
+--------------------------------
+
+Enabling encryption impacts performance. The performance degradation can vary
+based on the environment, queries, and concurrency.
+
+For queries that do not require transferring too much data between the Presto
+nodes (e.g. `SELECT count(*) FROM table`), the performance impact is negligible.
+
+However, for CPU intensive queries which require a considerable amount of data
+to be transferred between the nodes (for example, distributed joins, aggregations and
+window functions, which require repartitioning), the performance impact might be
+considerable. The slowdown may vary from 10% to even 100%+, depending on the network
+traffic and the CPU utilization.
+
+Advanced Performance Tuning
+---------------------------
+
+In some cases, changing the source of random numbers will improve performance
+significantly.
+
+By default, TLS encryption uses the `/dev/urandom` system device as a source of entropy.
+This device has limited throughput, so on environments with high network bandwidth
+(e.g. InfiniBand), powerful CPUs (multi-core, high-end server class) and high concurrency,
```

Comment:
So, the same configuration but with 10Gbps instead of Infiniband would not be an issue?
If yes, then let's remove everything after the InfiniBand to void confusion.